vital/types/camera_rpc.h

```diff
@@ -102,13 +103,17 @@ class VITAL_EXPORT camera_rpc : public camera
   /// Project a 2D image back to a 3D point in space
   virtual vector_3d back_project( const vector_2d& image_pt, double elev ) const;
 
+  /// Get an affine camera from the first order Taylor approximation
```

Comment:
This is a very interesting idea.  I'd be curious to know how well this performs.  Specifically, as you move away from the center point, how quickly does the approximation error increase.  It would be interesting to write a test case that uses one of the really RPC models from the RPC tests, picks an approximation point and 
 1) confirms that the projections are the same at the selected point
 2) computes the error (`(rpc.project(pt) - affine.project(pt)).norm()`) as you move the 3D point away from the selected point.

An alternative algorithm for approximating the affine camera would find the least squares solution over a volume, instead of at a single point.  That is,
 1) pick a bounding volume in world space
 2) sample a uniform 3D grid of points in this volume, say 100 x 100 x 10 (or maybe 10 x 10 x 10 is enough)
 3) project each grid point into the image space with the RPC model
 4) find a least squares estimate of the affine camera that maps the 3D points to the corresponding 2D points

This approach should distribute the error more evenly across that AOI.  However, I think your Taylor approximation solution has value too, so I wouldn't get rid of that.  It would be nice to compare the approaches.

Reply:
I did some experiments with the 2 methods. 
Here are the projection errors for a grid of 3d points (50x50 with a constant elevation), interpolated over the entire image. It was obtained for an image of AOI4 (~1.6km x 1.5km)

![heatmap](https://user-images.githubusercontent.com/36070340/47932061-e3ba4780-ded0-11e8-9ce2-84e0498406a0.png)

![cam_affine](https://user-images.githubusercontent.com/36070340/47932067-e87efb80-ded0-11e8-8485-cc269218e646.png)

Blue is Taylor approx. and orange is the least-square method.

You're right, the error is better distributed over the image and the global mean error is also slighty lower.

Taylor approx. mean error = 0.0850
Least-square method mean error = 0.0749

Reply:
Nice work.  Are these errors in units of pixels?  If so then either of these seems like a pretty good approximation.

Reply:
Yes, the errors are in pixels.